homeassistant/components/ihc/auto_setup.py

```diff
@@ -105,36 +108,47 @@ def autosetup_ihc_products(hass: HomeAssistant, config, ihc_controller, controll
     groups = project.findall(".//group")
     for platform in IHC_PLATFORMS:
         platform_setup = auto_setup_conf[platform]
-        discovery_info = get_discovery_info(platform_setup, groups, controller_id)
+        discovery_info = get_discovery_info(
+            platform_setup, groups, controller_id, use_groups
+        )
         if discovery_info:
             discovery.load_platform(hass, platform, DOMAIN, discovery_info, config)
 
     return True
 
 
-def get_discovery_info(platform_setup, groups, controller_id):
+def get_discovery_info(platform_setup, groups, controller_id, use_groups: bool):
     """Get discovery info for specified IHC platform."""
     discovery_data = {}
     for group in groups:
         groupname = group.attrib["name"]
         for product_cfg in platform_setup:
             products = group.findall(product_cfg[CONF_XPATH])
             for product in products:
+                product_id = int(product.attrib["id"].strip("_"), 0)
                 nodes = product.findall(product_cfg[CONF_NODE])
                 for node in nodes:
                     if "setting" in node.attrib and node.attrib["setting"] == "yes":
                         continue
                     ihc_id = int(node.attrib["id"].strip("_"), 0)
                     name = f"{groupname}_{ihc_id}"
+                    model = product.get("product_identifier") or ""
+                    # make the model number look a bit nicer - strip leading _
+                    if model.startswith("_"):
+                        model = model[1::]
```

Comment:
```suggestion
                    # make the model number look a bit nicer - strip leading _
                    model = product.get("product_identifier", "").lstrip("_")
```

Reply:
This may strip multiple _ prefixes but i assume that is OK

Reply:
It is just one _ prefix, but that is ok (and nicer).